caom2repo/caom2repo/core.py

```diff
@@ -539,6 +539,7 @@ def delete_observation(self, collection, observation_id):
         :param observation_id: ID of the observation
```

Comment:
whatever you choose in previous methods for whether or not to replace `observation_id` with `uri`, do that here too.